web_src/js/components/DiffFileTree.vue

```diff
@@ -29,6 +29,7 @@ export default {
   },
   computed: {
     fileTree() {
+      pageData.diffFileInfo.files = this.files;
```

Comment:
I haven't looked into the details, but I think it's improper to assign values in a `computed`  function, it's against the Vue's pattern?

Reply:
This is necessary to ensure that updates to the files in the pagedata causes updates to the sidebar files list

Reply:
While this works, it's a vue antipattern: https://vuejs.org/guide/essentials/computed.html#getters-should-be-side-effect-free

`pageData.diffFileInfo.files` should either be a computed as well or change based on a watcher (which could watch the `fileTree` computed.

Reply:
This is not mutating the value. It is not in that list of antipatterns.

It is simply exposing the data.files list on the pageData. That may well be an antipattern for vue but it is not listed on that page.

Reply:
I've moved it in to mounted as it doesn't need to be repeatedly done and to prove it is not a side-effect

Reply:
Moving it to `mounted` is better.

ps: although I have pointed out the similar problem in the original PR, but the original author didn't have interest to change,  it just "seemed" to work at that time,  so at that time I  just  made  some small changes  and approved.  In the end, no-good code causes problems.